ui/hooks/snaps/useNavigation.js

```diff
@@ -0,0 +1,22 @@
+import { useHistory } from 'react-router-dom';
+import { parseMetaMaskUrl } from '@metamask/snaps-utils';
+import { getSnapRoute } from '../../helpers/utils/util';
+
+const useNavigation = () => {
```

Comment:
This name seems too vague, something like `useSnapNavigation` would give more context

Reply:
The navigation isn't only to snap pages so that's why I chose to keep the name generic

Reply:
Yes but this hook is only useful in the snap context which in this case is custom UI. `useNavigation` seems like something you can use anywhere to navigate in the extension.